File: code-tree/intermediate-low/chapter2/lesson2/basic.cpp
#include <bits/stdc++.h>

using namespace std;

int k, n;

vector<int> temp;

vector<vector<int>> v;

void go(int cnt) {
    if(cnt == n){
        v.push_back(temp);
        return;
    }

    for(int i = 1; i <= k; i++){
        if(temp.size() >= 2 && temp[temp.size() - 2] == temp[temp.size() - 1] && temp[temp.size() - 1] == i) continue;
        temp.push_back(i);
        go(cnt + 1);
        temp.pop_back();
    }
}

int main() {
    cin >> k >> n;

    // Please write your code here.

    go(0);

    for(auto it : v){
        for(auto i : it){
            cout << i << " ";
        }
        cout << "\n";
    }

    return 0;
}

File: code-tree/intermediate-low/chapter2-backtracking/lesson4/basic.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);

// Please Write your code here.

const visited = Array(n + 1).fill(0);
const temp = [];

function go(cnt) {
  if(cnt === n){
    console.log(temp.join(' '))
    return;
  }

  for(let i = 1; i <= n; i++){
    if(visited[i]) continue;
    visited[i] = 1;
    temp.push(i);
    go(cnt + 1);
    visited[i] = 0;
    temp.pop();
  }
}

go(0);
